src/UI/UIUtils.java

```diff
@@ -335,15 +335,18 @@ public static COSpriteSpec fromDisk(String col, String fac)
   public static final COSpriteSpec LA = COSpriteSpec.fromDisk("Lazurian"      , "Army");
   public static final COSpriteSpec IDS = COSpriteSpec.fromDisk("IDS"          , "Army");
 
+  public static COSpriteSpec[] CANON_FACTIONS;
+  public static SourceGames[] GAME_RESOLUTION_ORDER;
+
   public enum SourceGames
   {
-    // In order of CO name discriminator resolution
-    AW3(GE,  "3"),
-    AW2(BM,  "2"),
+    // In display order, because programming is hard™
```

Comment:
Comment syntax error: unmatched quotation mark.

Reply:
Also I'm not sure I under stand the comment. This is the order in which we check faction portraits if the explicit one is not found. It doesn't line up with the `GAME_RESOLUTION_ORDER` below, which is what actually dictates the order in which the games are displayed on the CO selection screen.

Reply:
No that's backward. `GAME_RESOLUTION_ORDER` is used for the faction portraits. I'm not sure what the `SourceGames` ordering dictates, but the two aren't ordered the same.

Also it seems odd to have a separate enum array just to reorder the enum values.

Reply:
> Comment syntax error: unmatched quotation mark.

That's not a quotation mark.
> No that's backward.

Yes.
> `GAME_RESOLUTION_ORDER` is used for the faction portraits.

Yes. I guess I could call it `PORTRAIT_RESOLUTION_ORDER`, but that's even longer.
> I'm not sure what the `SourceGames` ordering dictates, but the two aren't ordered the same.

Display order on the CO selection screen.
> Also it seems odd to have a separate enum array just to reorder the enum values.

If you have a better solution, I'm all ears.
I thought about giving the enum values a portrait precedence value or something, but that's less clear and seems less efficient (need to sort the enums by a different ordering, or iterate over all of them).

Reply:
> That's not a quotation mark.

Oh, right. Guess I shoulda looked twice.

> I guess I could call it PORTRAIT_RESOLUTION_ORDER, but that's even longer.

Also more specific to how it's used.

> If you have a better solution, I'm all ears.

Given that it's preference-driven, and your preference for CO-selection-screen ordering is different from your preference for portrait ambiguity resolution, I think this is as good as anything.